src/application-insights/azext_applicationinsights/_params.py

```diff
@@ -4,17 +4,36 @@
 # --------------------------------------------------------------------------------------------
 
 # pylint: disable=line-too-long
-from azure.cli.core.commands.parameters import get_datetime_type
+from azure.cli.core.commands.parameters import get_datetime_type, get_location_type, tags_type
 from azure.cli.command_modules.monitor.actions import get_period_type
 from ._validators import validate_applications
 
 
 def load_arguments(self, _):
     with self.argument_context('monitor app-insights') as c:
         c.argument('application', options_list=['--app', '-a'], id_part='name', help='GUID, app name, or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal. If using an application name, please specify resource group.')
-        c.argument('start_time', arg_type=get_datetime_type(help='Start-time of time range for which to retrieve data.'))
-        c.argument('end_time', arg_type=get_datetime_type(help='End of time range for current operation. Defaults to the current time.'))
-        c.argument('offset', help='Filter results based on UTC hour offset.', type=get_period_type(as_timedelta=True))
+
+    with self.argument_context('monitor app-insights component create') as c:
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+        c.argument('application-type', options_list=['application-type', '--type', '-t'], help="Type of application being monitored. Possible values include: 'web', 'other'. Default value: 'web' .")
+        c.argument('kind', options_list=['--kind', '-k'], help='The kind of application that this component refers to, used to customize UI. This value is a freeform string, values should typically be one of the following: web, ios, other, store, java, phone.')
+        c.argument('tags', tags_type)
+
+    with self.argument_context('monitor app-insights component update') as c:
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+        c.argument('application-type', options_list=['application-type', '--type', '-t'], help="Type of application being monitored. Possible values include: 'web', 'other'. Default value: 'web' .")
```

Comment:
I assume you will get `--application-type` even w/o the leading `-`?

Reply:
aligned for consistency